cranelift/codegen/src/isa/x64/lower.isle

```diff
@@ -102,6 +102,19 @@
           (with_flags (x64_add_with_flags_paired $I64 x_lo y_lo)
                       (x64_adc_paired $I64 x_hi y_hi)))))
 
+;;;; Rules for `iadd_cout` ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+;; TODO: i8 and i16 support. Requires either learning how to encode ALU
+;; operations on values narrower than 32-bits (better code; big change) or doing
+;; the same extend-to-32-bits trick that aarch64 does (worse code; small
+;; change).
```

Comment:
What about restricting the carry operations to `i64`/`i32`?

Reply:
We could do that, but that is a CLIF semantics decision rather than an implementation decision (this comment is regarding the latter).